x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/RollupShardIndexer.java

```diff
@@ -172,17 +195,23 @@ public void beforeBulk(long executionId, BulkRequest request) {
             public void afterBulk(long executionId, BulkRequest request, BulkResponse response) {
                 numIndexed.addAndGet(request.numberOfActions());
                 if (response.hasFailures()) {
-                    Map<String, String> failures = Arrays.stream(response.getItems())
+                    List<BulkItemResponse> failedItems = Arrays.stream(response.getItems())
```

Comment:
I don't get why we need this change here. `failedItems.size()` should be equal to `failures.size()`. Isn't enough to just call `status.setCancelled()`?

Reply:
when the _id is null, the map key is the same, so the failures.size() is 1, so I use a list to same the failures